segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -323,20 +323,19 @@ public void updateSegmentAttribute(UpdateSegmentAttribute updateSegmentAttribute
         long trace = LoggerHelpers.traceEnter(log, operation, updateSegmentAttribute);
         val update = new AttributeUpdate(attributeId, AttributeUpdateType.ReplaceIfEquals, newValue, expectedValue);
         segmentStore.updateAttributes(segmentName, Collections.singletonList(update), TIMEOUT)
-                .whenComplete((v, e) -> {
-                    LoggerHelpers.traceLeave(log, operation, trace, e);
-                    if (e == null) {
-                        connection.send(new SegmentAttributeUpdated(requestId, true));
-                    } else {
-                        if (Exceptions.unwrap(e) instanceof BadAttributeUpdateException) {
-                            log.debug("Updating segment attribute {} failed due to: {}", update, e.getMessage());
-                            connection.send(new SegmentAttributeUpdated(requestId, false));
+                    .whenComplete((v, e) -> {
+                        LoggerHelpers.traceLeave(log, operation, trace, e);
+                        if (e == null) {
+                            connection.send(new SegmentAttributeUpdated(requestId, true));
                         } else {
-                            handleException(requestId, segmentName, operation, e);
+                            if (Exceptions.unwrap(e) instanceof BadAttributeUpdateException) {
```

Comment:
The point of the second "redundant" `handleException` is to make sure that the code in `whenComplete` doesn't throw any exceptions itself, thus leaving the connection hanging.

I recommend you put the contents of the `whenComplete` in a `try-catch` block and use `handleException` in the `catch`.

Reply:
The only place inside `whenComplete` where exceptions could be thrown is the `connection.send()`, which is because the connection is not present. 
I have updated the code to use the pattern`.thenAccept(..).exceptionally(...)`. Let me know if this is better.

Reply:
Updated the code with a `try-catch` block inside `whenComplete`.